### PR TITLE
Add Upwork Desktop Latest Version

### PR DESCRIPTION
**Pull request by bashu**
*If there’s a checkbox you can’t complete for any reason, that's okay, just explain in detail why you weren’t able to do so.*

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked that the cask was not already refused in [closed issues].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed

replacement of oDesk Team, ref #26539

### CODE REVIEW

**Comment by miccal** [comment]
I cannot see a link to this app from the homepage, I can only see a link to the [mobile app](https://www.upwork.com/i/mobile/)? Their [download page](https://www.upwork.com/downloads) appears to require a login?

**Comment by bashu** [comment]
@miccal suddenly yes! but you can download / install desktop app using this formula and no login required

**Comment by miccal** [comment]
Yes, but this may be classified as a [walled build](https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#unofficial-vendorless-and-walled-builds) which are not accepted for security reasons.

**Comment by bashu** [comment]
@miccal should I PR to https://github.com/alehouse/homebrew-unofficial instead?

**Comment by miccal** [comment]
Ping @vitorgalvao.

**Comment by bashu** [comment]
@miccal @vitorgalvao Upwork Desktop is a successor of oDesk Team which was a part of homebrew-cask in a past, ref #26539

File: Casks/upwork.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version :latest
+  sha256 :no_check
+
+  url 'https://updates-desktopapp.upwork.com/binaries/v4_2_88_0_zjqu5h1eh5fvxz5t/Upwork.dmg'
+  name 'Upwork'
+  homepage 'https://www.upwork.com/'
+
+  app 'Upwork.app'
+end
+~
```
Comment (vitorgalvao) [inline Casks/upwork.rb]: This definitely is not right.

File: Casks/upwork.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+cask 'upwork' do
+  version :latest
```
Comment (vitorgalvao) [inline Casks/upwork.rb]: `version '4_2_88_0_zjqu5h1eh5fvxz5t'`

File: Casks/upwork.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  sha256 :no_check
```
Comment (vitorgalvao) [inline Casks/upwork.rb]: `sha256` also needs to be set. You can find what it is by running `shasum -a 256` on the downloaded file.

File: Casks/upwork.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  url 'https://updates-desktopapp.upwork.com/binaries/v4_2_88_0_zjqu5h1eh5fvxz5t/Upwork.dmg'
```
Comment (vitorgalvao) [inline Casks/upwork.rb]: `url "https://updates-desktopapp.upwork.com/binaries/v#{version}/Upwork.dmg"`

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
Clarification regarding walled builds — see [the glossary](https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) (emphasis added):

> **Walled:** When the download URL is both behind a login/registration form **and from a host that differs from the homepage**.

That’s not the case, here. Issue is if you submit a casks with homepage `http://one.com` and the URL points to `http://two.com`. If the download URL is behind a registration/login wall, a user will have no easy way to know the download URL is trustworthy. If the download URL is from the same domain as the homepage, though, then you can be pretty sure the download is official (unless the website was hacked, but that would be an issue in every case).

**Comment by miccal** [comment]
Thanks for the clarification @vitorgalvao.

File: Casks/upwork.rb (hunk 2 above)
Comment (bashu) [inline Casks/upwork.rb]: @vitorgalvao fixed

File: Casks/upwork.rb (hunk 1 above)
Comment (bashu) [inline Casks/upwork.rb]: @vitorgalvao what an ugly copy-paste! fixed

**Comment by bashu** [comment]
@miccal @vitorgalvao thanks for feedback!

File: Casks/upwork.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+cask 'upwork' do
+  version '4_2_88_0_zjqu5h1eh5fvxz5t'
+  sha256 '4c321f521f2f42e0cd001a98274b99ea9b269ad8fc5334403c4065667ce56677'
+
+  url 'https://updates-desktopapp.upwork.com/binaries/v#{version}/Upwork.dmg'
```
Comment (miccal) [inline Casks/upwork.rb]: You need double quotes around the URL for the interpolation to work, like so:
```
url "https://updates-desktopapp.upwork.com/binaries/v#{version}/Upwork.dmg"
```

**Comment by bashu** [comment]
@miccal thank you

**Pull request merged**